aggregators/converter.go

```diff
@@ -43,21 +43,53 @@ func EventToCombinedMetrics(
 	e *modelpb.APMEvent,
 	unpartitionedKey CombinedMetricsKey,
 	partitioner Partitioner,
-) ([]CombinedKV, error) {
-	var (
-		kvs []CombinedKV
-		gl  GlobalLabels
-	)
+) (map[CombinedMetricsKey]*CombinedMetrics, error) {
+	var gl GlobalLabels
 	gl.fromLabelsAndNumericLabels(e.Labels, e.NumericLabels)
 	gls, err := gl.MarshalString()
 	if err != nil {
 		return nil, err
 	}
 
+	kvs := make(map[CombinedMetricsKey]*CombinedMetrics)
 	svcKey := serviceKey(e, unpartitionedKey.Interval)
 	svcInstanceKey := ServiceInstanceAggregationKey{GlobalLabelsStr: gls}
 	hasher := Hasher{}.Chain(svcKey).Chain(svcInstanceKey)
 
+	setCombinedMetrics := func(k CombinedMetricsKey, sim ServiceInstanceMetrics) {
```

Comment:
[**For reviewers**] This has gotten a bit messy. Suggestions to simplify this are welcomed (I am working on another PR to directly use the proto models instead of translating the models and I will try to simplify the code there).